### PR TITLE
Add context field validation in tag/collection forms

### PR DESCRIPTION
**Pull request by dmarkowicz**
I am targetting this branch, because it is BC.

## Changelog

```markdown
### Added
- context fields validation in tag/collection create forms, which prevents creating objects
  with empty context from UI
```

## Subject
Current create forms allow user to create objects with empty context, which is a bug (`sonata:classification:fix-context` command try to repair such situations).

This PR
- add `required` attribute and `NotBlank` constraint to tag/collection forms to show nice message to user.
- prevent creating tag/collection with no context from UI

### CODE REVIEW

**Comment by core23** [comment]
👎 For adding the context field to the edit page(s). You could create a parent child connection with different contexts which could cause problems. 

Please use the `getNewInstance` or `pre*` methods instead.

**Comment by dmarkowicz** [comment]
@core23 you're right, I forgot that context in tree should be the same. Removed category part from this PR for now. I will discuss category UI changes in another Issue.

File: Admin/CollectionAdmin.php [hunk 1]
```diff
@@ -53,7 +54,12 @@ protected function configureFormFields(FormMapper $formMapper)
             ->add('description', 'textarea', array(
                 'required' => false,
             ))
-            ->add('context')
+            ->add('context', null, array(
+                'constraints' => array(
```
Comment (core23) [inline Admin/CollectionAdmin.php]: Can we move this stuff to the `validation.xml` file?

Comment (dmarkowicz) [inline Admin/CollectionAdmin.php]: Done.

**Comment by OskarStark** [comment]
Thank you @dmarkowicz and sorry for the delay 👍

**Pull request merged**